pkg/apis/workflow/v1alpha1/types.go

```diff
@@ -153,6 +164,12 @@ type WorkflowSpec struct {
 	// Default scheduler will be used if neither specified.
 	// +optional
 	SchedulerName string `json:"schedulerName,omitempty"`
+
+	// PodReclaimation is the strategy when to delete the completed pods
+	// 1. Retain: always retain completed pods
+	// 2. Delete：delete pod immediately when it is completed
+	// 3. DeleteLater：delete completed pod when its workflow is completed
+	PodGCStrategy PodGCStrategy `json:"podgcstrategy,omitempty"`
```

Comment:
Instead of a string, I think the pod GC should be a struct so that we can expand on this in the future

```yaml
spec:
  podGC:
    strategy: OnCompletion
```

This will leave room do something like the following, which would delete successful pods after some time after they succeed:

```yaml
spec:
  podGC:
    strategy: OnSuccess
    delaySeconds: 120
```